### PR TITLE
Unsubscribing when browser permission is in a blocked or prompt statuses

### PR DESCRIPTION
**Pull request by k-antipochkin**
### Description of Changes

Whenever the browser permission observer catches a permissions blocked/prompt(in case of permissions reset) events, it performs an unsubscribe flow which also moves the push permissions to the 'unsubscribed' state.

### Breaking Changes

-   None

### Release Checklist

Prepare:

-   ~~[ ] Detail any breaking changes. Breaking changes require a new major version number~~

Bump versions in:

-   ~~[ ] package.json~~
-   ~~[ ] package-lock.json (run `npm install` after editing package.json)~~
-   ~~[ ] src/core/index.ts~~

Build:

-   [x] `npm run dist`
-   [x] Commit resulting `dist/` changes

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Fill out release notes
-   [ ] Run `npm publish --access public`
-   [ ] Publish `dist/worker.js` and `dist/web-bundle.js` to CDN (re-run `npm run dist` locally to update `web-bundle.js`)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -142,9 +142,9 @@ export default class Kumulos {
 
             if (permissionState === 'granted') {
                 this.pushManager.attemptPushRegister(this.context);
-            } else if (permissionState === 'denied') {
-                this.context.broadcastSubscriptionState(await this.pushManager.getCurrentSubscriptionState(this.context));
-            }
+            } else {
+                this.pushManager.pushUnsubscribe(this.context);
+            }
```
Comment (cgwyllie) [inline src/index.ts]: Formatting not applied?

File: src/index.ts [hunk 2]
```diff
@@ -142,9 +142,9 @@ export default class Kumulos {
 
             if (permissionState === 'granted') {
                 this.pushManager.attemptPushRegister(this.context);
-            } else if (permissionState === 'denied') {
-                this.context.broadcastSubscriptionState(await this.pushManager.getCurrentSubscriptionState(this.context));
-            }
+            } else {
```
Comment (cgwyllie) [inline src/index.ts:145]: Currently documented statuses are: https://developer.mozilla.org/en-US/docs/Web/API/PermissionStatus/state

So for denied & prompt we will unsubscribe, which seems good.

**Comment by cgwyllie** [comment]
In general changes LGTM, one query around an edge case.

Is it possible the user might not have a valid subscription to Optimove yet and changing browser state may raise an unregistered event to the backend? (this should be fine from backend perspective, just to check)

File: src/index.ts (hunk 1 above)
Comment (k-antipochkin) [reply src/index.ts]: Right, fixed

**Comment by k-antipochkin** [comment]
> In general changes LGTM, one query around an edge case.
> 
> Is it possible the user might not have a valid subscription to Optimove yet and changing browser state may raise an unregistered event to the backend? (this should be fine from backend perspective, just to check)

Yes, there was an assumption that the backend is ok with that

**Pull request merged**